feature/player/src/main/java/dev/anilbeesetti/nextplayer/feature/player/PlayerActivity.kt

```diff
@@ -289,6 +300,44 @@ class PlayerActivity : AppCompatActivity() {
         super.onStop()
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onUserLeaveHint() {
+        super.onUserLeaveHint()
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.R && playerPreferences.autoPip) {
+            this.enterPictureInPictureMode(updatePictureInPictureParams())
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onPictureInPictureModeChanged(isInPictureInPictureMode: Boolean, newConfig: Configuration) {
+        if (isInPictureInPictureMode) {
+            playerUnlockControls.visibility = View.INVISIBLE
+        } else {
+            if (!isControlsLocked) {
+                playerUnlockControls.visibility = View.VISIBLE
+            }
+        }
+        super.onPictureInPictureModeChanged(isInPictureInPictureMode, newConfig)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun updatePictureInPictureParams(): PictureInPictureParams {
+        var params: PictureInPictureParams = PictureInPictureParams.Builder()
+            .setAspectRatio(Rational(16, 9))
+            .build()
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            params = PictureInPictureParams.Builder()
+                .setAspectRatio(Rational(16, 9))
+                .setAutoEnterEnabled(playerPreferences.autoPip)
```

Comment:
I believe this is unnecessary because we are explicitly calling enterPictureInPictureMode()

Reply:
SetAutoEnabled has been set for animation for Android > S, As it is also recommended by Docs